mod_ood_proxy/lib/ood/nginx_stage.lua

```diff
@@ -74,12 +77,64 @@ end
 --[[
   capture2e
 
-  Give a string for stdin, get a string for merged stdout and stderr
+  Give a string for a command, get a string for merged stdout and stderr
 --]]
 function capture2e(cmd)
   return capture2(cmd .. " 2>&1")
 end
 
+--[[
+  capture2e_with_stdin
+
+  Give a command and some standard in, get a string for merged stdout and stderr
+--]]
+function capture2e_with_stdin(cmd, stdin)
+  math.randomseed(os.time())
+  local num = math.random(0,1000000000)
```

Comment:
not sure if this is random enough, but using `io.tmpfile` actually creates a real file, not a fifo.

Reply:
Oh I see. Maybe tmpfile is preferable then... https://unix.stackexchange.com/questions/29851/shell-script-mktemp-whats-the-best-method-to-create-temporary-named-pipe there is also `mktemp -u` and in Lua https://www.lua.org/manual/5.1/manual.html#pdf-os.tmpname but the latter still sometimes creates a file anyways.